File: app/src/main/java/com/merino/ddfilms/model/Review.java
package com.merino.ddfilms.model;


import com.google.firebase.firestore.Exclude;
import com.merino.ddfilms.utils.DateFormatter;

import java.util.List;

import lombok.Data;

@Data
public class Review {
    private String id;
    private String userId;
    private String userName;
    private String userProfileImageUrl;
    private int movieId;
    private String movieTitle;
    private String posterPath;
    private String backdropPath;
    private float rating; // De 0.5 a 5.0 (incrementos de 0.5)
    private String reviewText;
    private String reviewDate;
    private List<String> likeCount;
    private List<String> dislikeCount;
    private transient boolean isLikedByCurrentUser;
    private transient boolean isDislikedByCurrentUser;

    @Exclude
    public boolean isLikedByCurrentUser() {
        return isLikedByCurrentUser;
    }

    @Exclude
    public void setLikedByCurrentUser(boolean likedByCurrentUser) {
        isLikedByCurrentUser = likedByCurrentUser;
    }

    @Exclude
    public boolean isDislikedByCurrentUser() {
        return isDislikedByCurrentUser;
    }

    @Exclude
    public void setDislikedByCurrentUser(boolean dislikedByCurrentUser) {
        isDislikedByCurrentUser = dislikedByCurrentUser;
    }

    @Exclude
    public String getFormattedDate() {
        DateFormatter formatter = new DateFormatter();
        return formatter.getFormattedDate(this.reviewDate);
    }

}

File: app/src/main/java/com/merino/ddfilms/model/Review.kt
package com.merino.ddfilms.model

import com.google.firebase.firestore.Exclude
import com.merino.ddfilms.utils.DateFormatter

data class Review(
    var id: String? = null,
    var userId: String? = null,
    var userName: String? = null,
    var userProfileImageUrl: String? = null,
    var movieId: Int = 0,
    var movieTitle: String? = null,
    var posterPath: String? = null,
    var backdropPath: String? = null,
    var rating: Float = 0f,
    var reviewText: String? = null,
    var reviewDate: String? = null,
    var likeCount: List<String>? = null,
    var dislikeCount: List<String>? = null
) {
    @get:Exclude
    @set:Exclude
    var isLikedByCurrentUser: Boolean = false

    @get:Exclude
    @set:Exclude
    var isDislikedByCurrentUser: Boolean = false

    @get:Exclude
    val formattedDate: String
        get() {
            val formatter = DateFormatter()
            return formatter.getFormattedDate(this.reviewDate)
        }
}
